lib/improver/ensemble_calibration/ensemble_calibration_utilities.py

```diff
@@ -94,3 +94,113 @@ def check_predictor_of_mean_flag(predictor_of_mean_flag):
+        """Initialise the plugin.
+
+        Args:
+            historic_forecast_dict (dict):
+                Dictionary specifying the metadata that defines the historic
+                forecast. For example:
+                ::
+
+                    {
+                        "attributes": {
+                            "mosg__model_configuration": "uk_ens"
+                        }
+                    }
+            truth_dict (dict):
+                Dictionary specifying the metadata that defines the truth.
+                For example:
+                ::
+
+                    {
+                        "attributes": {
+                            "mosg__model_configuration": "uk_det"
+                        }
+                    }
+
+        Raises:
+            NotImplementedError:
+                'attributes' is the only supported key for the input
+                dictionaries.
+        """
+        self.historic_forecast_dict = historic_forecast_dict
+        self.truth_dict = truth_dict
+
+        for input_dict in [self.historic_forecast_dict, self.truth_dict]:
+            for key in input_dict.keys():
+                if key != "attributes":
+                    msg = ("At present, 'attributes' is the only supported "
+                           "key for the input. Support for specifying other "
+                           "metadata can be added if required.")
+                    raise NotImplementedError(msg)
+
+    def __repr__(self):
+        """Represent the plugin instance as a string."""
+        result = ('<SplitHistoricForecastsAndTruth: '
+                  'historic_forecast_dict={}, truth_dict={}>')
+        return result.format(self.historic_forecast_dict, self.truth_dict)
+
+    @staticmethod
+    def _find_required_cubes_using_metadata(cubes, input_dict):
+        """
+        Extract the cube that matches the information within the input
+        dictionary.
+
+        Args:
+            cubes (iris.cube.CubeList):
+                The cubes that will be checked for matches against the
+                metadata specified in the input_dict.
+            input_dict (dict):
+                A dictionary containing the metadata that will be used to
+                identify the desired cubes.
+
+        Returns:
+            iris.cube.CubeList:
+                CubeList containing cubes that match the metadata supplied
+                within the input dictionary.
+
+        Raises:
+            ValueError:
+                The metadata supplied resulted in no matching cubes.
+
+        """
+        constr = iris.AttributeConstraint(**input_dict["attributes"])
+        cubelist = cubes.extract(constr)
+        if not cubelist:
+            msg = ("The metadata to identify the desired historic forecast or "
+                   "truth has found nothing matching the metadata information "
+                   "supplied: {}".format(input_dict))
+            raise ValueError(msg)
```

Comment:
Need to think about if this is desired behaviour from a suite perspective. Should this be a warning?

Reply:
After discussing this, this exception has been kept but this exception is converted into a warning within the CLI, as there would be instances, for example for the first day when initially creating a training dataset, the truth datasets will accumulate, however, there won't be a valid historic forecast until one day after the initial cycle, as we're only using historic forecasts with the same cycle point within our calibration.